packages/react-components/react-nav-preview/src/components/NavCategoryItem/NavCategoryItem.types.ts

```diff
@@ -10,20 +10,23 @@ export type NavCategoryItemSlots = {
   /**
    * The root element
    */
-  root: Slot<'button'>;
+  root: NonNullable<Slot<'button'>>;
 
   /**
-   * The component to be used as button in heading
+   * Icon that renders before the content.
```

Comment:
If the expand icon is able to render before or after the content, shouldn't this be able to as well? If we have the chevron on the left, it would make more sense to have the folder icon on the right otherwise you have icon clutter.